### PR TITLE
Add contributor ladder process.

### LINKED ISSUE #448: community: Is there a contributor ladder?
# community: Is there a contributor ladder? 

Opening this to ask if there's a contributor ladder defined for this project.
Here's an example: https://github.com/kubernetes/community/blob/master/community-membership.md

Subtext: I'd be happy to help do PR review here, with the hopes of working towards maintainership, if there's a path for that :)

## Previous PRs

1. https://github.com/guacsec/guac/pull/436
2. https://github.com/guacsec/guac/pull/433
3. https://github.com/guacsec/guac/pull/415
4. https://github.com/guacsec/guac/pull/392
5. https://github.com/guacsec/guac/pull/386
6. https://github.com/guacsec/guac/pull/378
7. https://github.com/guacsec/guac/pull/377
8. https://github.com/guacsec/guac/pull/356
9. https://github.com/guacsec/guac/pull/351
10. https://github.com/guacsec/guac/pull/348
11. https://github.com/guacsec/guac/pull/338
12. https://github.com/guacsec/guac/pull/337
13. https://github.com/guacsec/guac/pull/335
14. https://github.com/guacsec/guac/pull/334
15. https://github.com/guacsec/guac/pull/330

## Open PRs 
1. https://github.com/guacsec/guac/pull/390
2. https://github.com/guacsec/guac/pull/406
3. https://github.com/guacsec/guac/pull/408
4. https://github.com/guacsec/guac/pull/409
5. https://github.com/guacsec/guac/pull/416
6. https://github.com/guacsec/guac/pull/434
7. https://github.com/guacsec/guac/pull/435
8. https://github.com/guacsec/guac/pull/437
9. https://github.com/guacsec/guac/pull/438
10. https://github.com/guacsec/guac/pull/446

## Issues opened
1. https://github.com/guacsec/guac/issues/331
2. https://github.com/guacsec/guac/issues/379
3. https://github.com/guacsec/guac/issues/407
4. https://github.com/guacsec/guac/issues/410
5. https://github.com/guacsec/guac/issues/439
6. https://github.com/guacsec/guac/issues/329
7. https://github.com/guacsec/guac/issues/336
8. https://github.com/guacsec/guac/issues/339
9. https://github.com/guacsec/guac/issues/350
10. https://github.com/guacsec/guac/issues/381

cc @lumjjb @pxp928 @mihaimaruseac @mlieberman85

### PR DESCRIPTION
**Pull request by mihaimaruseac**
Should fix #448

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -4,7 +4,9 @@ We'd love to accept your patches and contributions to this project. There are
 just a few small guidelines you need to follow.
 
 Refer to the [Developer Setup](./SETUP.md) document to learn how to prepare your
-environment.
+environment. You can also consult a [GraphQL Guide](./demo/GraphQL.md) which
+replaces the Neo4j components of `SETUP.md`. Optionally, you can also consult
```
Comment (lumjjb) [inline CONTRIBUTING.md:8]: i think we are going to remove `SETUP.md` so this shouldn't be necessary anymore

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -40,3 +49,105 @@ labels
...
+- Front-end: GUAC visualisation libraries, Javascript, CSS
+- Clients: GUAC CLI, CI Checks, Policy Engines, IDE Plugins, etc.
+
+The contribution ladder is summarized in the table below:
+
+|         Role          | Responsibilities | Requirements | Privileges | GitHub access level |
+|-----------------------|------------------|--------------|------------|---------------------|
+| Community Participant | Follow [Google's Open Source Community Guidelines](https://opensource.google/conduct/) | N/A | N/A | Read (default) |
+| Reviewer              | Review contributions from other members | Prove technical expertise in at least one GUAC topic area | Can approve PRs | Write |
+| Owner                 | Set direction and priorities for topic area. Approval of contributions | Deep technical expertise in topic area. Proven record of reviews and authorship. | Slack channel invitation | Write |
```
Comment (lumjjb) [inline CONTRIBUTING.md:75]: we don't own slack so cant put much in slack channel invitation here, maybe google groups?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -40,3 +49,105 @@ labels
...
+owners and/on maintainers can propose graduating the community participant to
+the next role, that of Reviewer.
+
+A reviewer will be responsible of reviewing contributions from other
+contributors that touch the corresponding topic area. They should ensure that
+the PRs are adequately tested, follow the guidelines for the specific topic
+area.
+
+Since reviewer gain write access to the repository, they can also contribute
+directly, without needing to create a fork.
```
Comment (lumjjb) [inline CONTRIBUTING.md:101]: hmm, i know probably not the best practice, but i keep doing it out of convenience. i think we should definitely still encourage folks to use forks so as not to have a danging branches problem (which CNCF TAG security repo suffers from).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -40,3 +49,105 @@ labels
...
+Each role in the contribution ladder requires maintaining active contributions
+at the corresponding level. Inactivity is harmful to the project, so maintainers
+can decide to remove privileges from inactive contributors, after a majority
+vote. Inactivity is defined as periods longer than 3 months during which no
+contribution is being done.
+
+Contributors also have the option of stepping down voluntarily, by contacting
+the maintainers when circumstances affect future contribution potential. For
+highly active contributors that completely step away from the project, we will
+create an emeritus process.
```
Comment (lumjjb) [inline CONTRIBUTING.md:151]: +1 , great addition here on emeritus process, want to keep the knowledge around.

**Review by lumjjb — APPROVED** [review]
Nice! This is looking great!! Thanks @mihaimaruseac !

File: CONTRIBUTING.md (hunk 3 above)
Comment (pxp928) [reply CONTRIBUTING.md:101]: +1

File: CONTRIBUTING.md (hunk 2 above)
Comment (mihaimaruseac) [reply CONTRIBUTING.md]: Made it more general

File: CONTRIBUTING.md (hunk 1 above)
Comment (mihaimaruseac) [reply CONTRIBUTING.md]: Added a comment to that end, thinking we would update this when we delete SETUP.md

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -26,6 +29,13 @@ use GitHub pull requests for this purpose. Consult
 [GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
 information on using pull requests.
 
+We require all commits in a PR to contain a `Signed-off-by` line which can be
+added by using the `-s` flag of `git commit`.
```
Comment (jeffmendoza) [inline CONTRIBUTING.md:33]: We shouldn't simply tell folks to do this without telling them what it means:
https://developercertificate.org/
https://wiki.linuxfoundation.org/dco

Probably better to link out than summarize.

Comment (mihaimaruseac) [reply CONTRIBUTING.md:33]: Right. Added a link to the wiki.

**Review by mlieberman85 — APPROVED** [review]
Lgtm

**Pull request merged**